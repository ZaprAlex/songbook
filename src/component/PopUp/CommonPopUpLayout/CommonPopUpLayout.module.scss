.content {
    display: flex;
    flex-flow: column nowrap;
    overflow: hidden;
    padding: 48px;

    @media (width < 576px) {
        padding: 24px;
    }
}

.title {
    width: 100%;
    margin-bottom: 16px;
    font-size: 32px;
    line-height: 1.25;
    letter-spacing: -0.3px;
    text-align: left;
    white-space: pre-wrap;
    color: #000000;

    @media (width < 576px) {
        margin-bottom: 8px;
        font-size: 22px;
        line-height: 30px;
        letter-spacing: 0.2px;
    }

    .centered > & {
        text-align: center;
    }
}

.footer {
    width: 100%;
    margin-top: 32px;
    display: inline-flex;
    text-align: left;

    @media (width < 576px) {
        margin-top: 16px;
    }

    .centered > & {
        justify-content: center;
    }
}
