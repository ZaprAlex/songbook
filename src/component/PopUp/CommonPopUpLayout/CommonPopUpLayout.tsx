import React, { FC } from 'react';
import cn from 'classnames';

import Button from '../../Button';

import styles from './CommonPopUpLayout.module.scss';

export type CommonPopUpLayoutProps = {
    onClick: () => void;
    title?: string;
    buttonLabel?: string;
    centered?: boolean;
};

const CommonPopUpLayout: FC<CommonPopUpLayoutProps> = ({
    onClick,
    title,
    buttonLabel = 'Понятно',
    centered = false,
    children,
}) => (
    <div className={cn(styles.content, { [styles.centered]: centered })}>
        {title && <p className={styles.title}>{title}</p>}
        {children}
        <div className={styles.footer}>
            <Button
                label={buttonLabel}
                onClick={onClick}
                type="contained"
                skeuomorph
                size="middle"
                dataTestId="information-close-button"
            />
        </div>
    </div>
);

export default CommonPopUpLayout;
