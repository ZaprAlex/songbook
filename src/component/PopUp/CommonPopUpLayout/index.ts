import CommonPopUpLayout from './CommonPopUpLayout';

export default CommonPopUpLayout;
