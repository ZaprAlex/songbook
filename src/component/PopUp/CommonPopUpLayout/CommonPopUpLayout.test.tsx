import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import CommonPopUpLayout from './CommonPopUpLayout';

describe('CommonPopUpLayout', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
    });
    const onClick = jest.fn();
    const closeButtonDataTestId = 'information-close-button';
    const childrenMock = 'pseudoChildren';
    const testTitle = 'testTitle';
    const testButtonLabel = 'testButtonLabel';

    const component = () =>
        render(
            <CommonPopUpLayout onClick={onClick} title={testTitle} buttonLabel={testButtonLabel}>
                {childrenMock}
            </CommonPopUpLayout>
        );

    it('render with default props', () => {
        const { container } = render(<CommonPopUpLayout onClick={onClick} />);

        expect(container.firstChild).toMatchSnapshot();
        expect(screen.getByText('Понятно')).toBeInTheDocument();
    });

    it('render all', () => {
        const { container } = component();

        expect(container.firstChild).toMatchSnapshot();
        expect(screen.getByText(testTitle)).toBeInTheDocument();
        expect(screen.getByText(childrenMock)).toBeInTheDocument();
        expect(screen.getByText(testButtonLabel)).toBeInTheDocument();
    });

    it('test close btn', () => {
        component();

        const informationCloseButton = screen.getByTestId(closeButtonDataTestId);
        expect(informationCloseButton).toBeInTheDocument();

        fireEvent.click(informationCloseButton);
        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
