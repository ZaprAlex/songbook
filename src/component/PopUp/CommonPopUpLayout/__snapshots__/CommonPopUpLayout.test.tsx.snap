// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonPopUpLayout render all 1`] = `
<div
  class="content"
>
  <p
    class="title"
  >
    testTitle
  </p>
  pseudoChildren
  <div
    class="footer"
  >
    <button
      class="btn btn-green btn-contained btn-middle btnSkeuomorph"
      data-test-id="information-close-button"
      type="button"
    >
      <span
        class="buttonLabel"
      >
        testButtonLabel
      </span>
    </button>
  </div>
</div>
`;

exports[`CommonPopUpLayout render with default props 1`] = `
<div
  class="content"
>
  <div
    class="footer"
  >
    <button
      class="btn btn-green btn-contained btn-middle btnSkeuomorph"
      data-test-id="information-close-button"
      type="button"
    >
      <span
        class="buttonLabel"
      >
        Понятно
      </span>
    </button>
  </div>
</div>
`;
